docs/build.rst

```diff
@@ -1,117 +1,344 @@
+the build will fail. If the requirement is satisfied the module is applied to the image object.
 
-**Buildah and Podman options are:**
+The last step is to copy only required modules (module repository can contain many modules)
+from the temporary directory to the final target directory.
 
-* ``--pull`` -- ask a builder engine to check and fetch latest base image
-* ``--tag`` -- an image tag used to build image (can be specified multiple times)
-* ``--help`` -- Show this message and exit.
+Handling artifacts
+************************
 
-**Docker options are:**
+Each module and image descriptor itself can define :ref:`artifacts <descriptor/image:Artifacts>`.
 
-* ``--pull`` -- ask a builder engine to check and fetch latest base image
-* ``--tag`` -- an image tag used to build image (can be specified multiple times)
-* ``--no-squash`` -- do not squash the image after build is done.
-* ``--help`` -- Show this message and exit.
+In this step CEKit is going to handle all defined artifacts for the image. For each defined
+artifact CEKit is going to fetch it. If there will be a problem while fetching the artifact,
+CEKit will fail with information why it happened.
 
-**OSBS options are:**
+Each successfully fetched artifact is automatically added to :doc:`cache</caching>` so that
+subsequent build will be executed faster without the need to download the artifact again.
 
-* ``--release`` -- perform a OSBS release build
-* ``--tech-preview`` -- updates image descriptor ``name`` key to contain ``-tech-preview`` suffix in family part of the image name
-* ``---user`` -- alternative user passed to `rhpkg --user`
-* ``--nowait`` -- run `rhpkg container-build` with `--nowait` option specified
-* ``--stage`` -- use ``rhpkg-stage`` tool instead of ``rhpkg``
-* ``--koji-target`` -- overrides the default ``koji`` target
-* ``--commit-msg`` -- custom commit message for dist-git
-* ``--help`` -- Show this message and exit.
+Generating required files
+******************************
 
+When we have all external content handled and the image object is final we can generate required files.
+Generation is tightly coupled with the selected builder because different builders require different
+files to be generated.
 
-Docker build
-^^^^^^^^^^^^^^^^
+For example Docker builder requires Dockerfile to be generated, but the OSBS builder requires
+additional files besides the Dockerfile.
 
-This is the default way to build an container image. The image is build utilizing Docker daemon via Python binding.
+For Dockerfiles we use a template which is populated which can access the image object properties.
 
-**Example:** Building a docker image
+Build execution
+************************
 
-.. code:: bash
+Final step is to execute the build using selected builder.
 
-	  $ cekit build
+Resulting image sometimes will be available on your localhost, sometimes in some remote
+registry. It all depends on the builder.
 
+Common ``build`` parameters
```

Comment:
Should ``build`` be in code style ? - or have it all in heading style ?

Reply:
I think it should be without code style, I'll change it.